### PR TITLE
More audit todos and refactoring

### PR DESCRIPTION
**Pull request by rlan35**
local test passed.

### CODE REVIEW

File: consensus/votepower/roster.go [hunk 1]
```diff
@@ -154,58 +154,58 @@ func (r *Roster) JSON() string {
...
 		r.TheirVotingPowerTotalPercentage.String(),
-		r.RawStakedTotal.String(),
+		r.TotalEffectiveStake.String(),
 	}
 	b, _ := json.Marshal(&c)
 	return string(b)
 }
 
 // Compute creates a new roster based off the shard.SlotList
+// TODO(audit): the decimal computation is memory and cpu intensive.
```
Comment (fxfactorial) [inline consensus/votepower/roster.go]: This compute happens once

**Comment by fxfactorial** [comment]
Actually there is lots of overlap of implentqtion with my current PR

File: consensus/votepower/roster.go (hunk 1 above)
Comment (rlan35) [reply consensus/votepower/roster.go]: it happens in accumulate rewards every block

**Comment by fxfactorial** [comment]
Ya I addressed that issue in my Pr

**Comment by fxfactorial** [comment]
Some of these TODOs were subsumed in the APR PR, this open issue leaves two optimization TODOs left to do: https://github.com/harmony-one/harmony/issues/2529

**Pull request closed without merging**